samples/onnx_accelerator_sample1/onnxacceleratorsampleone/main_stack.py

```diff
@@ -129,18 +131,85 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+                                                  }
+                                                  )
+      
+      function_iot_deployment.add_to_role_policy(iam.PolicyStatement(
+        effect=iam.Effect.ALLOW,
+        actions=[
+          'greengrass:ListComponents',
+          'greengrass:GetDeployment',
+          'greengrass:CreateDeployment',
+          'greengrass:ListDeployments',
+          'greengrass:ListComponentVersions'
+
+        ],
+        resources=[
+          'arn:aws:greengrass:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':components:*',
+          'arn:aws:greengrass:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':/greengrass/groups/*',
+          'arn:aws:greengrass:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':deployments',
+          'arn:aws:greengrass:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':deployments:*'
+        ]
+      ))
+
+      function_iot_deployment.add_to_role_policy(iam.PolicyStatement(
+        effect=iam.Effect.ALLOW,
+        actions=[
+          'iot:DescribeJob',
+          'iot:CancelJob',
+          'iot:CreateJob'
+
+        ],
+        resources=[
+          'arn:aws:iot:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':job/*',
+          'arn:aws:iot:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':thinggroup/*'
+        ]
+      ))
+    else:
+      # attach the lambda which will be triggered everytime there is a new object created
+      function_iot_deployment = _lambda.Function(self, "lambda_function",
+                                                  runtime=_lambda.Runtime.PYTHON_3_9,
+                                                  handler="lambda.handler",
+                                                  code=_lambda.Code.from_asset("functions/iotjobcreator/src"),
+                                                  function_name="iot_job_deployment",
+                                                  environment={
+                                                      'THING_GROUP_NAME': iot_thing_group_name,
+                                                      'ARN_IOT_PROVISIONING_ROLE': iot_job_s3_role.role_arn
+                                                  }
+                                                  )
+
+      function_iot_deployment.add_to_role_policy(iam.PolicyStatement(
+        effect=iam.Effect.ALLOW,
+        actions=[
+          'iam:PassRole'
+        ],
+        resources=[
+          'arn:aws:iam::'+ Aws.ACCOUNT_ID+':role/'+iot_job_s3_role.role_name
+        ]
+      ))
+
+      function_iot_deployment.add_to_role_policy(iam.PolicyStatement(
+        effect=iam.Effect.ALLOW,
+        actions=[
+          'iot:CreateJob'
+        ],
+        resources=[
+          'arn:aws:iot:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':job/*',
+          'arn:aws:iot:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':thing/*',
+          'arn:aws:iot:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':thinggroup/*',
+          'arn:aws:iot:'+ Aws.REGION+':'+ Aws.ACCOUNT_ID+':jobtemplate/*'
+        ]
+      ))
+
```

Comment:
Why don't you create the additional policies needed for installing Greengrass on the RPi and the policy to access the S3 bucket for the TokenExchangeRole?